tendermint-rs/src/config.rs

```diff
@@ -0,0 +1,619 @@
+    /// and verifying their commits
+    pub fast_sync: bool,
+
+    /// Database backend: `leveldb | memdb | cleveldb`
+    pub db_backend: DbBackend,
+
+    /// Database directory
+    pub db_dir: PathBuf,
+
+    /// Output level for logging, including package level options
+    pub log_level: LogLevel,
+
+    /// Output format: 'plain' (colored text) or 'json'
+    pub log_format: LogFormat,
+
+    /// Path to the JSON file containing the initial validator set and other meta data
+    pub genesis_file: PathBuf,
+
+    /// Path to the JSON file containing the private key to use as a validator in the consensus protocol
+    pub priv_validator_key_file: Option<PathBuf>,
+
+    /// Path to the JSON file containing the last sign state of a validator
+    pub priv_validator_state_file: PathBuf,
+
+    /// TCP or UNIX socket address for Tendermint to listen on for
+    /// connections from an external PrivValidator process
+    #[serde(deserialize_with = "deserialize_optional_value")]
+    pub priv_validator_laddr: Option<net::Address>,
+
+    /// Path to the JSON file containing the private key to use for node authentication in the p2p protocol
+    pub node_key_file: PathBuf,
+
+    /// Mechanism to connect to the ABCI application: socket | grpc
+    pub abci: AbciMode,
+
+    /// TCP or UNIX socket address for the profiling server to listen on
+    #[serde(deserialize_with = "deserialize_optional_value")]
+    pub prof_laddr: Option<net::Address>,
+
+    /// If `true`, query the ABCI app on connecting to a new peer
+    /// so the app can decide if we should keep the connection or not
+    pub filter_peers: bool,
+
+    /// rpc server configuration options
+    pub rpc: RpcConfig,
+
+    /// peer to peer configuration options
+    pub p2p: P2PConfig,
+
+    /// mempool configuration options
+    pub mempool: MempoolConfig,
+
+    /// consensus configuration options
+    pub consensus: ConsensusConfig,
+
+    /// transactions indexer configuration options
+    pub tx_index: TxIndexConfig,
+
+    /// instrumentation configuration options
+    pub instrumentation: InstrumentationConfig,
+}
+
+impl TendermintConfig {
+    /// Parse Tendermint `config.toml`
+    pub fn parse_toml<T: AsRef<str>>(toml_string: T) -> Result<Self, Error> {
+        Ok(toml::from_str(toml_string.as_ref())?)
+    }
+
+    /// Load `config.toml` from a file
+    pub fn load_toml_file<P>(path: &P) -> Result<Self, Error>
+    where
+        P: AsRef<Path>,
+    {
+        let toml_string = fs::read_to_string(path).map_err(|e| {
+            err!(
+                ErrorKind::Parse,
+                "couldn't open {}: {}",
+                path.as_ref().display(),
+                e
+            )
+        })?;
+
+        Self::parse_toml(toml_string)
+    }
+
+    /// Is this configuration for a validator?
+    pub fn is_validator(&self) -> bool {
+        self.priv_validator_key_file.is_some() || self.priv_validator_laddr.is_some()
```

Comment:
default tendermint config has a validator key file but isn't necessarily for a validator